airflow/hooks/webhdfs_hook.py

```diff
@@ -0,0 +1,81 @@
+from airflow.hooks.base_hook import BaseHook
+from airflow.configuration import conf
+import logging
+
+from hdfs import InsecureClient, HdfsError
+from airflow.utils import AirflowException
+
+
+class WebHDFSHookException(AirflowException):
+    pass
+
+
+class WebHDFSHook(BaseHook):
+    """
+    Interact with HDFS. This class is a wrapper around the hdfscli library.
+    """
+    def __init__(self, webhdfs_conn_id='webhdfs_default'):
+        self.webhdfs_conn_id = webhdfs_conn_id
+
+    def get_conn(self):
+        """
+        Returns a hdfscli InsecureClient object.
+        """
+        nn_connections = self.get_connections(self.webhdfs_conn_id)
+        for nn in nn_connections:
+            try:
+                logging.debug('Trying namenode {}'.format(nn.host))
+                client = InsecureClient('http://{nn.host}:{nn.port}'.format(nn=nn))
```

Comment:
Alright, I added a micron of support for this. I have no ability to test this other that it does not interfere with using the InsecureClient on our setup. Let me know if you get the chance to test it. The API provided by the extra seems [fairly basic](http://hdfscli.readthedocs.org/en/latest/api.html#module-hdfs.ext.kerberos), I do not know if it has all you need.